src/app/Plans/components/helpers.ts

```diff
@@ -1,33 +1,133 @@
+import { ProgressVariant } from '@patternfly/react-core';
 import { IKubeList } from '@app/client/types';
-import { PlanStatusType, PlanStatusDisplayType } from '@app/common/constants';
+import { PlanStatusDisplayType, WarmPlanState, PlanState } from '@app/common/constants';
 import { hasCondition } from '@app/common/helpers';
 import { isSameResource } from '@app/queries/helpers';
-import { IPlan } from '@app/queries/types';
+import { IPlan, IStatusCondition } from '@app/queries/types';
 import { IMigration } from '@app/queries/types/migrations.types';
 import { UseQueryResult } from 'react-query';
+import { PlanActionButtonType } from '@app/Plans/components/PlansTable';
 
 export const getPlanStatusTitle = (plan: IPlan): string => {
   const condition = plan.status?.conditions.find(
     (condition) =>
-      condition.type === PlanStatusType.Ready ||
-      condition.type === PlanStatusType.Executing ||
-      condition.type === PlanStatusType.Succeeded ||
-      condition.type === PlanStatusType.Failed
+      condition.type === 'Ready' ||
+      condition.type === 'Executing' ||
+      condition.type === 'Succeeded' ||
+      condition.type === 'Failed'
   );
   return condition ? PlanStatusDisplayType[condition.type] : '';
 };
 
-// TODO maybe generalize this for cold migrations too
-type WarmPlanState =
-  | 'NotStarted'
-  | 'Starting'
-  | 'Copying'
-  | 'AbortedCopying'
-  | 'StartingCutover'
-  | 'Cutover'
-  | 'Finished';
+export const getMigStatusState = (plan: IPlan) => {
```

Comment:
Can this be driven from the `PlanState` as well instead of directly from `hasCondition` cases? If they don't map 1:1 we may want to add/change the states of a plan so that they do. I see you have a TODO below to add different types of Finished states, e.g. Canceled. That may be worth doing before we introduce duplicate logic. (I guess there already is some duplicate logic... we could do that in a followup PR if you want)

Reply:
Yep, totally makes sense - I'll go ahead and do it as part of this PR